mqtt/mqtt-bridge/src/rpc.rs

```diff
@@ -0,0 +1,503 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+
+use bson::{doc, Document};
+use bytes::{buf::BufExt, Bytes};
+use lazy_static::lazy_static;
+use mqtt3::{
+    proto::Publication, proto::QoS, proto::SubscribeTo, Event, PublishError,
+    SubscriptionUpdateEvent,
+};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use tracing::{error, warn};
+
+#[cfg(test)]
```

Comment:
q: why do you need this here?

Reply:
for tests, I want to pass mocks into event handler code. for production code I want real implementation to be used